l1-contracts/contracts/governance/PermanentRestriction.sol

```diff
@@ -183,4 +235,51 @@ contract PermanentRestriction is IRestriction, IPermanentRestriction, Ownable2St
             revert NotAnAdmin(admin, _potentialAdmin);
         }
     }
+
+    /// @notice Tries to get the new admin from the migration.
+    /// @param _call The call data.
+    /// @dev This function reverts if the provided call was not a migration call.
+    function tryGetNewAdminFromMigration(Call calldata _call) external view returns (address) {
+        if (_call.target != address(BRIDGE_HUB)) {
+            revert NotBridgehub(_call.target);
+        }
+
+        if (bytes4(_call.data[:4]) != IBridgehub.requestL2TransactionTwoBridges.selector) {
```

Comment:
Why it can be `requestL2TransactionDirect` or why it can be a direct call to the `Mailbox` for Era?